source/magnifier/magnifier.py

```diff
@@ -0,0 +1,300 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2025 NV Access Limited, Antoine Haffreingue
+# This file may be used under the terms of the GNU General Public License, version 2 or later, as modified by the NVDA license.
+# For full terms and any additional permissions, see the NVDA license file: https://github.com/nvaccess/nvda/blob/master/copying.txt
+
+from typing import Callable
+from logHandler import log
+import wx
+import api
+import winUser
+import mouseHandler
+from winAPI._displayTracking import getPrimaryDisplayOrientation
+from .utils.types import (
+	MagnifierPosition,
+	Coordinates,
+	MagnifierType,
+	Direction,
+	FocusType,
+	Filter,
+)
+
+from .config import getDefaultZoomLevel, getDefaultFilter, ZoomLevel
+
+
+class Magnifier:
+	_ZOOM_MIN: float = ZoomLevel.MIN_ZOOM
+	_ZOOM_MAX: float = ZoomLevel.MAX_ZOOM
+	_ZOOM_STEP: float = ZoomLevel.STEP_FACTOR
+	_TIMER_INTERVAL_MS: int = 20
+	_MARGIN_BORDER: int = 50
+	display = getPrimaryDisplayOrientation()
+	_SCREEN_WIDTH: int = display.width
+	_SCREEN_HEIGHT: int = display.height
```

Comment:
this can change over time if users change monitor setups.
It should be set in the init function and possibly updated by registering an action to the `displayChanged` extension point